infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureToleranceDecisionEngine.java

```diff
@@ -0,0 +1,54 @@
+package org.corfudb.infrastructure.management;
+
+import org.corfudb.protocols.wireprotocol.NodeView;
+import org.corfudb.runtime.view.Layout;
+
+import java.util.List;
+
+/**
+ * {@link FailureToleranceDecisionEngine} provides methods to decide about failure or healing
+ * status of Corfu servers in a given {@link Layout} and for a specific view of the cluster
+ * captured in a {@link NodeView}.
+ * Decisions are dependant on the concrete {@link FailureDiscoveryPolicy} which underlies the
+ * concrete implementations of this interface.
+ *
+ * Created by Sam Behnam on 10/19/18.
+ */
+public interface FailureToleranceDecisionEngine {
```

Comment:
The name is something i am also struggling with. I do not have a good suggestion though

Reply:
The name indicates that the engine is meant (to the extent of underlying algorithm) to tolerate the failures. For instance its decision could tolerate a link failure not removing a node when based on the specific policy it is not necessary to do so.